src/productcatalogservice/Dockerfile

```diff
@@ -11,16 +11,27 @@ COPY Gopkg.* ./
 RUN dep ensure --vendor-only -v
 
 COPY . .
-RUN go build -o /productcatalogservice .
+RUN go build -gcflags=all='-N -l' -o /productcatalogservice .
 
 FROM alpine AS release
-RUN apk add --no-cache ca-certificates
+RUN apk add --no-cache ca-certificates \
+            libc6-compat
+
 RUN GRPC_HEALTH_PROBE_VERSION=v0.2.0 && \
     wget -qO/bin/grpc_health_probe https://github.com/grpc-ecosystem/grpc-health-probe/releases/download/${GRPC_HEALTH_PROBE_VERSION}/grpc_health_probe-linux-amd64 && \
     chmod +x /bin/grpc_health_probe
+
 WORKDIR /productcatalogservice
+
+RUN wget -O go-cloud-debug https://storage.googleapis.com/cloud-debugger/compute-go/go-cloud-debug \
+            && chmod 0755 go-cloud-debug
+
 COPY --from=builder /productcatalogservice ./server
 COPY products.json .
+COPY source-context.json .
```

Comment:
you can merge them as

    COPY products.json source-context.json .

also I'm not sure if this will error out if source-context.json is missing.

We should be able to support deployments that don't require users to generate debugging-related files. If docker fails to run this line when it's missing, you might need to use a wildcard like `source-context*.json`.